packages/component-library/components/Heading/Heading.tsx

```diff
@@ -0,0 +1,79 @@
+import classNames from "classnames"
+import { createElement } from "react"
+
+const styles = require("./Heading.module.scss")
+
+export type HeadingVariants =
+  | "display-0"
+  | "heading-1"
+  | "heading-2"
+  | "heading-3"
+  | "heading-4"
+  | "heading-5"
+  | "heading-6"
+
+export type AllowedTags =
+  | "pre"
+  | "p"
+  | "a"
+  | "div"
+  | "span"
+  | "h1"
+  | "h2"
+  | "h3"
+  | "h4"
+  | "h5"
+  | "h6"
+
+export interface HeadingProps {
+  classNameAndIHaveSpokenToDST?: string
+  children: React.ReactNode
```

Comment:
I'm sure I've seen headings in the wild that contain things like italic for just one word, so making this flexible witih `ReactNode` is a probably a good thing.

I'm not so familiar with the TS react typings - Does `React.ReactNode`  also permit a list of nodes?

Eg:
`<h1>The <em>final</em> countdown </h1>`
or would you have to do:
`<h1><div>The <em>final</em> countdown</div></h1>`

Reply:
Here's the definition for the types: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L203

So all of those examples you posted would be valid with this component.